pandas/_typing.py

```diff
@@ -1,33 +1,35 @@
 from pathlib import Path
-from typing import IO, AnyStr, TypeVar, Union
+from typing import IO, TYPE_CHECKING, AnyStr, TypeVar, Union
 
 import numpy as np
 
-from pandas._libs import Timestamp
-from pandas._libs.tslibs.period import Period
-from pandas._libs.tslibs.timedeltas import Timedelta
+if TYPE_CHECKING:  # Use for any internal imports
+    from pandas._libs import Timestamp
+    from pandas._libs.tslibs.period import Period
+    from pandas._libs.tslibs.timedeltas import Timedelta
```

Comment:
Timedelta and Period can also come directly from _libs; not sure if mypy cares about that